mod_ci/controllers.py

```diff
@@ -598,14 +598,22 @@ def progress_reporter(test_id, token):
                 status = TestStatus.from_string(request.form['status'])
                 # Check whether test is not running previous status again
                 istatus = TestStatus.progress_step(status)
+                message = request.form['message']
                 if len(test.progress) != 0:
                     laststatus = TestStatus.progress_step(
                         test.progress[-1].status)
                     if laststatus > istatus:
                         status = TestStatus.canceled
-
+                        message = "Duplicate Entries"
+                        platform = test.platform
+                        if platform == TestPlatform.linux:
+                            kvm_name = config.get('KVM_LINUX_NAME', '')
+                            shut_down_vm(kvm_name)
```

Comment:
There's still a chance that between you call it and shut it down, it's posting another one. It's easier to store the canceled message (as you did in the first commit), and then simply return wihtout storing in subsequent requests until the VM is powered down.